src/Databases/DataLake/tests/test_hive_catalog_url_parsing.cpp

```diff
@@ -0,0 +1,160 @@
+#include <Databases/DataLake/HiveCatalog.h>
+#include <gtest/gtest.h>
+#include <Common/Exception.h>
+#include <base/types.h>
+
+namespace DB::ErrorCodes
+{
+extern const int DATALAKE_DATABASE_ERROR;
+}
+
+namespace DB::DataLake::Test
+{
+
+class HiveCatalogUrlParsingTest : public ::testing::Test
+{
+protected:
+    void SetUp() override {}
+    void TearDown() override {}
+};
+
+std::pair<String, Int32> testParseHostPort(const String & url)
+{
+    size_t protocol_sep = url.find("://");
+    if (protocol_sep == String::npos)
+        throw DB::Exception(DB::ErrorCodes::DATALAKE_DATABASE_ERROR, "Invalid URL format: missing protocol separator '://'");
+
+    size_t start = protocol_sep + 3;
+
+    size_t colon_pos = url.find(':', start);
+    if (colon_pos == String::npos)
+        throw DB::Exception(DB::ErrorCodes::DATALAKE_DATABASE_ERROR, "Invalid URL format: missing port number");
+
+    size_t slash_pos = url.find('/', start);
+
+    String host = url.substr(start, colon_pos - start);
+
+    size_t port_end = (slash_pos != String::npos) ? slash_pos : url.size();
+    String port_str = url.substr(colon_pos + 1, port_end - colon_pos - 1);
+
+    if (port_str.empty() || !std::all_of(port_str.begin(), port_str.end(), ::isdigit))
+        throw DB::Exception(DB::ErrorCodes::DATALAKE_DATABASE_ERROR, "Invalid port number: '{}'", port_str);
+
+    try
+    {
+        Int32 port = std::stoi(port_str);
+        if (port <= 0 || port > 65535)
+            throw DB::Exception(DB::ErrorCodes::DATALAKE_DATABASE_ERROR, "Port number out of valid range (1-65535): {}", port);
+        return {host, port};
+    }
+    catch (const std::out_of_range&)
+    {
+        throw DB::Exception(DB::ErrorCodes::DATALAKE_DATABASE_ERROR, "Port number too large: {}", port_str);
```

Comment:
Here you should throw the exception message

Reply:
now the exception message is - Invalid port number format: {}